src/components/Product/FilterSidebar/FilterCheckboxGroup.tsx

```diff
@@ -1,39 +1,45 @@
-import { useMemo, useState } from 'react'
+import React, { useEffect, useState } from 'react'
 import Checkbox from '@/components/UI/Checkbox/Checkbox'
 import Button from '@/components/UI/Buttons/Button/Button'
 import Image from 'next/image'
+import FiltersGroupTitle from '@/components/Product/FilterSidebar/FiltersGroupTitle'
 
 interface IFilterCheckboxGroup {
   selectedItems: string[]
   items: string[]
   onFilterCheckboxClick: (item: string) => void
+  onReset: () => void
   title: string
 }
 
 const FilterCheckboxGroup = ({
   selectedItems,
   items,
-  onFilterCheckboxClick,
+  onFilterCheckboxClick = () => {},
+  onReset = () => {},
   title = '',
-}: IFilterCheckboxGroup) => {
+}: Readonly<IFilterCheckboxGroup>) => {
   const [isItemsToggled, setIsItemsToggled] = useState(true)
+  const [displayedItems, setDisplayedItems] = useState(() => items.slice(0, 5))
 
-  const displayedItems = useMemo(() => {
+  useEffect(() => {
     if (isItemsToggled) {
-      return items.slice(0, 5)
+      setDisplayedItems(items.slice(0, 5))
+
+      return
     }
 
-    return items
-  }, [items, isItemsToggled])
+    setDisplayedItems(items)
+  }, [isItemsToggled, items])
 
   const toggleItemsButtonClick = () => {
     setIsItemsToggled((prev) => !prev)
   }
```

Comment:
Maybe we can just use function 'toggleItemsButtonClick' without 'useEffect'.
like this:
 ```
 const toggleItemsButtonClick = () => {
    if (isItemsToggled) {
      setIsItemsToggled(false)
      setDisplayedItems(items)
    }
    else {
      setIsItemsToggled(true)
      setDisplayedItems(items.slice(0, 5))
    }
  }
```
  Without 'useEffect' there will be 1 less render.

Reply:
fixed in https://github.com/Sunagatov/Iced-Latte-Frontend/pull/219